src/libutil/posix-source-accessor.cc

```diff
@@ -110,12 +139,15 @@ SourceAccessor::DirEntries PosixSourceAccessor::readDirectory(const CanonPath &
 std::string PosixSourceAccessor::readLink(const CanonPath & path)
 {
     if (auto parent = path.parent()) assertNoSymlinks(*parent);
-    return nix::readLink(path.abs());
+    std::filesystem::path target { nix::readLink(makeAbsPath(path)) };
+    if (target.is_absolute())
+        target = root / std::move(target);
```

Comment:
I don't think prepending `root` is correct. Since this is a virtual filesystem rooted at `root`, if we then try to `readFile()` the result of `readLink()`, `root` will be prepended *again*. I think we don't want `root` to ever leak out.

Reply:
Yes this does seem wrong, but was how the `readLink` was working before.

I figure this *is* correct if the read link absolute path was supposed to be relative the conceptual root, but not sure if that happens in practice. (e.g. when unpacking a git repo I think the symlinks are already adjusted, if absolute ones are even supported.)

Reply:
Hm what was wrong about that? It passes through the result from `readlink` unmodified, which is the correct behaviour. In general symlinks shouldn't be interpreted by `readLink` at all, it's up to the caller to decide what to do with relative/absolute symlinks.

Reply:
Oh right, just the input

Reply:
Sorry I got all confused, I will just delete that stuff.